File: 品味Java的21種設計模式/dp_cpp/bridge/example1/CommonMessageEmail.cpp
#include "CommonMessageEmail.h"
#include <iostream>

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace bridge
			{
				namespace example1
				{

					void CommonMessageEmail::send(std::string message, std::string toUser)
					{
                        std::cout << "使用Email的方式，发送消息'" << message << "给" << toUser;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/bridge/example1/CommonMessageEmail.java
package cn.javass.dp.bridge.example1;
/**
 * 以Email的方式发送普通消息
 */
public class CommonMessageEmail implements Message{

	public void send(String message, String toUser) {
		System.out.println("使用Email的方式，发送消息'"+message+"'给"+toUser);
	}

}